libs/wingc/src/ast.rs

```diff
@@ -184,6 +209,16 @@ pub struct ElifBlock {
 	pub statements: Scope,
 }
 
+#[derive(Debug)]
+pub struct Class {
+	pub name: Symbol,
+	pub members: Vec<ClassMember>,
+	pub methods: Vec<(Symbol, FunctionDefinition)>,
+	pub constructor: Constructor,
+	pub parent: Option<Type>,
```

Comment:
This won't work, since a `Type` is a name (in this case a `UserDefinedType`) and a `Class` is a definition of a class (init,fields,methods...). The `parent` field should be the **name** of the parent type not the parent type definition itself.
It **might** be possible to change this to a  `UserDefinedType` I'll give it a try.

Reply:
>  It might be possible to change this to a UserDefinedType I'll give it a try.

done.